minigu/storage/src/olap/olap_graph.rs

```diff
@@ -0,0 +1,2176 @@
+    pub label_id: Option<LabelId>,
+    pub src_id: VertexId,
+    pub dst_id: VertexId,
+    pub properties: OlapPropertyStore,
+}
+
+// Olap-Property (Add 'Option' for compaction)
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Default)]
+pub struct OlapPropertyStore {
+    properties: Vec<Option<ScalarValue>>,
+}
+
+#[allow(dead_code)]
+impl OlapPropertyStore {
+    pub fn set_prop(&mut self, index: usize, prop: Option<ScalarValue>) {
+        self.properties.insert(index, prop);
+    }
+
+    pub fn get(&self, index: usize) -> Option<ScalarValue> {
+        self.properties.get(index).cloned().flatten()
+    }
+
+    pub(crate) fn new(properties: Vec<Option<ScalarValue>>) -> OlapPropertyStore {
+        OlapPropertyStore { properties }
+    }
+}
+
+// Block of edge array (Header + Actual Storage + MVCC)
+#[derive(Clone, Debug)]
+#[allow(dead_code)]
+pub struct EdgeBlock {
+    // Locate the previous block of the same vertex
+    pub pre_block_index: Option<usize>,
+    pub cur_block_index: usize,
+    pub is_tombstone: bool,
+    // Min and max edge id (Eid)
+    // For accelerating get_edge
+    pub max_label_id: Option<LabelId>,
+    pub min_label_id: Option<LabelId>,
+    // Min and max to id (However may not be used)
+    pub max_dst_id: VertexId,
+    pub min_dst_id: VertexId,
+    // Edge storage
+    pub src_id: VertexId,
+    pub edge_counter: usize,
+    pub edges: [OlapStorageEdge; BLOCK_CAPACITY],
+}
+
+// Edge block after compression
+#[derive(Clone, Debug)]
+#[allow(dead_code)]
+pub struct CompressedEdgeBlock {
+    // Locate the previous block of the same vertex
+    pub pre_block_index: Option<usize>,
+    pub cur_block_index: usize,
+    // Min and max edge id (Eid)
+    // For accelerating get_edge
+    pub max_label_id: Option<LabelId>,
+    pub min_label_id: Option<LabelId>,
+    // Min and max to id (Vid)
+    pub max_dst_id: VertexId,
+    pub min_dst_id: VertexId,
+    // Edge storage
+    pub src_id: VertexId,
+    pub edge_counter: usize,
+    pub delta_bit_width: u8,
+    pub first_dst_id: VertexId,
+    pub compressed_dst_ids: BitVec<u64, Lsb0>,
+    pub label_ids: [Option<LabelId>; BLOCK_CAPACITY],
+}
+
+// Property block (Column storage)
+#[derive(Clone, Debug)]
+pub struct PropertyBlock {
+    /// Property storage
+    pub values: Vec<Option<ScalarValue>>,
+}
+// Property column storage
+#[derive(Debug)]
+pub struct PropertyColumn {
+    pub blocks: Vec<PropertyBlock>,
+}
+
+// Property block after compaction
+#[derive(Clone, Debug)]
+#[allow(dead_code)]
+pub struct CompressedPropertyBlock {
+    pub bitmap: BitVec<u16, Lsb0>,
+    // Stands for numbers not null elements in every 16 elements
+    pub offsets: [u8; BLOCK_CAPACITY / 16],
+    pub values: Vec<ScalarValue>,
+}
+// Property column after compaction
+#[derive(Debug)]
+pub struct CompressedPropertyColumn {
+    pub blocks: Vec<CompressedPropertyBlock>,
+}
+
+// Graph storage for Olap (CSR)
+#[allow(dead_code)]
+pub struct OlapMvccGraphStorage {
```

Comment:
this struct can be `OlapStorage` or `OlapGraph`. More concise. Mvcc is a feature not part of the name.